devito/petsc/types/object.py

```diff
@@ -160,9 +182,21 @@ def fields(self):
 
     @property
     def time_dim_fields(self):
+        """
+        Fields within the struct that are updated during the time loop.
+        These are not set in the `PopulateMatContext` callback.
+        """
         return [f for f in self.fields
                 if isinstance(f, (ModuloDimension, TimeDimension))]
```

Comment:
`if isinstance(f, (ModuloDimension, TimeDimension))]`
                could be replaced with
`if f.is_Time`
?